packages/blade/src/utils/useTruncationTooltip/useTruncationTooltip.web.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+const useTruncationTooltip = ({
+  content,
+}: {
+  content?: string;
+}): {
+  containerRef: React.RefObject<HTMLDivElement>;
+  textRef: React.RefObject<HTMLParagraphElement>;
+} => {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const textRef = React.useRef<HTMLParagraphElement>(null);
+
+  React.useEffect(() => {
```

Comment:
Should we use layoutEffect?

Reply:
doesn't have to be render-blocking so should be fine for now. Can change this if its causing bugs